drivers/onewire/onewire.py

```diff
@@ -46,23 +47,25 @@ def select_rom(self, rom):
         self.reset()
         self.writebyte(MATCH_ROM)
         self.write(rom)
-
-    def scan(self):
+        
+    # Scan bus roms or alarm roms. Argument <cmd> shall take SERCH_ROM (default) or ALARM_SEARCH constants.
+    def scan(self, cmd=SEARCH_ROM):
         devices = []
         diff = 65
         rom = False
         for i in range(0xff):
-            rom, diff = self._search_rom(rom, diff)
+            rom, diff = self._search_rom(rom, diff, cmd)
             if rom:
                 devices += [rom]
             if diff == 0:
                 break
         return devices
-
-    def _search_rom(self, l_rom, diff):
+    
+    # Search bus roms or alarm roms. Argument <cmd> shall take SERCH_ROM (default) or ALARM_SEARCH constants.
```

Comment:
Typo: `SERCH_ROM`

Reply:
Tomorrow I'll try to finish to program Commands to Recall and Copy Tl,Th and config register between Scratchpad and EEPROM to avoid rewrite resolution and alarm each time start uC.